plugins/device/rest.go

```diff
@@ -87,35 +86,48 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	err = txn.Ctx.Set("provisioner", &req.Provisioner)
+	if err != nil {
+		logger.WithError(err).WithField("key", "provisioner").Error("failed to set key in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
 	err = txn.Do()
 	if err != nil {
 		logger.WithError(err).Error("Transaction to prepare device failed")
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "transaction to prepare device failed")
 		return
 	}
-	peerInfo, err = peer.GetPeer(peerID)
+	deviceInfo, err := deviceutils.GetDevice(peerID, req.Device)
 	if err != nil {
-		logger.WithError(err).WithField("peerid", peerID).Error("Failed to get peer from store")
-		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get peer from store")
+		logger.WithError(err).WithField("peerid", peerID).Error("Failed to get device from store")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get device from store")
 		return
 	}
 
 	// FIXME: Change this to http.StatusCreated when we are able to set
 	// location header with a unique URL that points to created device.
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, peerInfo)
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, deviceInfo)
 }
 
 func deviceListHandler(w http.ResponseWriter, r *http.Request) {
 
 	ctx := r.Context()
 	logger := gdctx.GetReqLogger(ctx)
 	peerID := mux.Vars(r)["peerid"]
-	if uuid.Parse(peerID) == nil {
+	if peerID != "" && uuid.Parse(peerID) == nil {
```

Comment:
If the peerID is empty then uuid.Parse() will return nil, so IMO we don't have to put an additional check.